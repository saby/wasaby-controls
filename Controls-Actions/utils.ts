import { IContextValue } from 'Controls/context';
import { Record } from 'Types/entity';

export interface IContextData extends IContextValue {
    documentContext?: Record;
}

/**
 * Получает объект из контекста
 * @public
 * @param context
 * @param objectName
 */
function findObject(context: IContextData, objectName: string): unknown {
    return context?.[objectName];
}

export { findObject };
