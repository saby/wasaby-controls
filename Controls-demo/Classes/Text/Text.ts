import { Control, IControlOptions, TemplateFunction } from 'UI/Base';
import controlTemplate = require('wml!Controls-demo/Classes/Text/Text');
import 'css!Controls/CommonClasses';

class ViewModes extends Control<IControlOptions> {
    protected _template: TemplateFunction = controlTemplate;
}
export default ViewModes;
