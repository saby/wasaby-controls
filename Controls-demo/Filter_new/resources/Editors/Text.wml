<Controls.input:Text
   on:valueChanged="valueChangedHandler()"
   value="{{_options.item.value}}"
/>
