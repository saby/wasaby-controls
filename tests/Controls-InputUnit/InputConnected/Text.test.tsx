/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { WasabyEvents } from 'UICore/Events';
import { render } from '@testing-library/react';
import { getContent } from '../resources/utils';
import { default as Constraint } from 'Controls-Input-demo/InputConnected/Text/Constraint';
import { default as DefaultValue } from 'Controls-Input-demo/InputConnected/Text/DefaultValue';
import { default as Label } from 'Controls-Input-demo/InputConnected/Text/Label';
import { default as Length } from 'Controls-Input-demo/InputConnected/Text/Length';
import { default as Multiline } from 'Controls-Input-demo/InputConnected/Text/Multiline';

describe(
    'Controls-Input/InputConnected:Text',
    () => {
        setTestID(148321705);

        let container = null;

        beforeEach(() => {
            jest.useFakeTimers('modern');
            jest.setSystemTime(new Date(1996, 5, 21));
            container = document.createElement('div');
            WasabyEvents.initInstance(container);
            document.body.appendChild(container);
        });
        afterEach(() => {
            unmountComponentAtNode(container);
            WasabyEvents.destroyInstance(container);
            jest.useRealTimers();
            container.remove();
            container = null;
        });

        it('defaultValue', async () => {
            render(getContent(DefaultValue), {container});
            expect(container).toMatchSnapshot();
        });
        it('Constraint', async () => {
            render(getContent(Constraint), {container});
            expect(container).toMatchSnapshot();
        });
        it('label', async () => {
            render(getContent(Label), {container});
            expect(container).toMatchSnapshot();
        });
        it('Length', async () => {
            render(getContent(Length), {container});
            expect(container).toMatchSnapshot();
        });
        it('Multiline', async () => {
            render(getContent(Multiline), {container});
            expect(container).toMatchSnapshot();
        });
    }
);
