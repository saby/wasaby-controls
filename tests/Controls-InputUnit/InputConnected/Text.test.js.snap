// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls-Input/InputConnected:Text Constraint 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            constraint = onlyLetters
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
            />
            <div
              class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            constraint = notSpecial
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
            />
            <div
              class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            constraint is empty
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
            />
            <div
              class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Text Length 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col  controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            length
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default"
        maxlength="15"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="Дважды в год лета не бывает."
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            length is empty
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
            />
            <div
              class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Text Multiline 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col  controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            multiline
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretEmpty controls-Render_multiline controls-Area controls-Area_line-height_m controls-max-w-full tw-flex-grow"
        style="--calculated-line-min-height_inputArea: calc(3 * var(--line-height_m_inputArea)); --calculated-line-min-height_inputArea-scroll: calc(3 * var(--line-height_m_inputArea) + (2 * var(--padding-vertical_inputArea))); --calculated-line-max-height_inputArea-scroll: calc(5 * var(--line-height_m_inputArea) + (2 * var(--padding-vertical_inputArea)));"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper controls-Render__wrapper_footer"
        >
          <div
            class="controls-Scroll-Container controls-Scroll controls_scroll_theme-default controls-Area__scroll controls-InputBase__field controls-Area__minHeight_countLines_3_size_m_indented controls-Area__maxHeight_countLines_5_size_m_indented controls-Render__field controls-Render__field_textAlign_left controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            style="min-height: var(--calculated-line-min-height_inputArea-scroll); max-height: var(--calculated-line-max-height_inputArea-scroll);"
            tabindex="0"
          >
            <div
              class="controls-Scroll-ContainerBase controls_scroll_theme-default controls-Scroll__content  controls-Scroll__content_hideNativeScrollbar controls-Scroll__content_hideNativeScrollbar_ff-ie-edge controls-Scroll-ContainerBase__scroll_vertical controls-Scroll-ContainerBase__scrollPosition-regular controls-Scroll-Container__base controls-BlockLayout__blockGroup undefined"
              data-qa="controls-Scroll__content"
              name="content"
              tabindex="0"
            >
              <div
                class="controls-Scroll-ContainerBase__content controls-Scroll-ContainerBase__content__vertical controls-Scroll-ContainerBase__content-direction_column"
                name="userContent"
              >
                <div
                  class="controls-Area__fieldWrapper controls-Area__field_margin-null controls-Scroll-containerBase_userContent"
                  name="fieldWrapper"
                  tabindex="0"
                >
                  <div
                    class="controls-Area__fakeField controls-Area__minHeight_countLines_3_size_m"
                    name="fakeField"
                    style="min-height: var(--calculated-line-min-height_inputArea);"
                  >
                    Дважды в год лета не бывает.﻿
                  </div>
                  <textarea
                    class="controls-Field js-controls-Field controls-InputBase__nativeField controls-Area__realField controls-InputBase__nativeField_caretFilled"
                    name="ws-input_1996-06-20"
                    spellcheck="true"
                    tabindex="0"
                    wrap="hard"
                  >
                    Дважды в год лета не бывает.
                  </textarea>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col  controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            multiline is empty
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="Дважды в год лета не бывает."
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Text defaultValue 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth400 tw-flex tw-flex-col"
  >
    <div
      class="tw-flex controls-max-w-full tw-flex-col  controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            defaultValue = 2
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="so good!!!"
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col "
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            defaultValue is empty
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
            />
            <div
              class="controls-InputBase__placeholder controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Controls-Input/InputConnected:Text label 1`] = `
<div>
  <div
    class="controlsDemo__wrapper controlsDemo_fixedWidth300 tw-flex tw-flex-col"
  >
    <div
      class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default controls-margin_top-m controls-margin_top-m controls-Input-connected_filled tw-w-full"
      tabindex="0"
    >
      <div
        class="controls-Render__wrapper"
      >
        <span
          class="controls-Render__baseline"
        >
          ﻿
        </span>
        <div
          class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
          data-qa="controls-Render__field"
          tabindex="0"
        >
          <input
            autocapitalize="off"
            autocomplete="off"
            autocorrect="off"
            class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
            inputmode="text"
            name="ws-input_1996-06-20"
            spellcheck="true"
            tabindex="0"
            type="text"
            value="Дважды в год лета не бывает."
          />
          <div
            class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
          >
            <div
              class="controls-Render__placeholder controls-Render__placeholder_overflow"
            >
              Введите текст
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="controls-JumpingLabel controls-margin_top-m controls-Input-connected_filled"
    >
      <div
        class="ws-flexbox ws-flex-column"
      >
        <div
          class="controls-JumpingLabel__wrapper"
        >
          <div
            class="controls-JumpingLabel__value controls-JumpingLabel__value_valid
               "
          >
            jumping
          </div>
        </div>
        <div
          class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default controls-JumpingLabel__content"
          tabindex="0"
        >
          <div
            class="controls-Render__wrapper"
          >
            <span
              class="controls-Render__baseline"
            >
              ﻿
            </span>
            <div
              class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis"
              data-qa="controls-Render__field"
              tabindex="0"
            >
              <input
                autocapitalize="off"
                autocomplete="off"
                autocorrect="off"
                class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
                inputmode="text"
                name="ws-input_1996-06-20"
                spellcheck="true"
                tabindex="0"
                type="text"
                value="Дважды в год лета не бывает."
              />
              <div
                class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-above-caret"
              >
                <div
                  class="controls-Render__placeholder controls-Render__placeholder_overflow"
                >
                  <div
                    class="controls-JumpingLabel__placeholder-wrapper"
                  >
                    <div
                      class="controls-JumpingLabel__placeholder"
                    >
                      jumping
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-col controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            label top
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default controls-margin_top-m"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="Дважды в год лета не бывает."
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m controls-Input-connected_filled"
    >
      <a
        class="controls-Label controls-Label_underline_none tw-min-w-0 tw-flex-shrink"
      >
        <div
          class="controls-Label__wrapper"
        >
          <div
            class="controls-Label__caption controls-fontsize-undefined controls-Label__caption_fontsize-default controls-Label__caption_fontColorStyle-label controls-Label__caption_underline_none controls-Label__caption_underline_none_edit ws-ellipsis"
          >
            label start
          </div>
          
        </div>
      </a>
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default controls-input_label_min-input-content controls-margin_top-m"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="Дважды в год лета не бывает."
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="tw-flex controls-max-w-full tw-flex-row tw-items-baseline controls-margin_top-m controls-Input-connected_filled"
    >
      <span
        class="controls-icon_size-s controls-icon_style-secondary icon-SabyBird controls-icon controls-margin_right-s"
      />
      <div
        class="controls-Render js-controls-Render controls-Render_border-top-empty controls-Render_border-bottom controls-Render_background-same controls-Render_textAlign-left controls-Render_borderRadius controls-Render_state-valid controls-fontsize-m controls-fontsize-m controls-fontweight-default controls-Render-fontsize-m controls-text-default controls-Render_state-valid_caretFilled controls-inlineheight-default controls-Render-inlineheight-default controls-Text controls_Text_theme-default controls-margin_top-m"
        tabindex="0"
      >
        <div
          class="controls-Render__wrapper"
        >
          <span
            class="controls-Render__baseline"
          >
            ﻿
          </span>
          <div
            class="controls-InputBase__field  controls-InputBase__field_margin-null controls-InputBase__field_theme_default_margin-null controls-Render__field controls-Render__field_textAlign_left ws-ellipsis controls-Render__field_zIndex"
            data-qa="controls-Render__field"
            tabindex="0"
          >
            <input
              autocapitalize="off"
              autocomplete="off"
              autocorrect="off"
              class="controls-Field js-controls-Field controls-InputBase__nativeField controls-InputBase__nativeField_caretFilled controls-InputBase__nativeField_caretFilled_theme_default"
              inputmode="text"
              name="ws-input_1996-06-20"
              spellcheck="true"
              tabindex="0"
              type="text"
              value="Дважды в год лета не бывает."
            />
            <div
              class="controls-InputBase__placeholder ws-hidden controls-InputBase__placeholder_displayed-under-caret"
            >
              <div
                class="controls-Render__placeholder controls-Render__placeholder_overflow"
              >
                Введите текст
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
